bool lighthouse_done
bool flag_done
